pkg/operators/machine-api-operator.go

```diff
@@ -65,6 +66,34 @@ var _ = Describe("[Feature:Operators] Machine API operator deployment should", f
 		Expect(framework.IsDeploymentAvailable(client, maoManagedDeployment, framework.MachineAPINamespace)).To(BeTrue())
 
 	})
+
+	It("reconcile termination handler daemonSet", func() {
+		client, err := framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		initialDaemonSet, err := framework.GetDaemonSet(client, terminationHandlerDaemonSet, framework.MachineAPINamespace)
+		Expect(err).NotTo(HaveOccurred())
+
+		By(fmt.Sprintf("checking daemonSet is available"))
+		Expect(framework.IsDaemonSetAvailable(client, terminationHandlerDaemonSet, framework.MachineAPINamespace)).To(BeTrue())
```

Comment:
What does available here mean? Does it mean that all replicas are running? If so, on a default cluster, the daemonset should always be available by virtue of it having no replicas. I think we need to simulate somewhere in the test suite that the daemonset is available and has more than 1 replica. I'm not really sure what this is testing over the daemonset just existing

Perhaps we do that in https://github.com/openshift/cluster-api-actuator-pkg/blob/master/pkg/infra/spot.go#L73? Could follow up later.

Reply:
This is validating the operator does its job and the expectation of having no unavailable replicas is satisfiedhttps://github.com/openshift/cluster-api-actuator-pkg/pull/177/files#diff-a8166de82f0b6261e02122357a0c6096R40.
On a default cluster the expected available happens to be zero. That's circumstantial, this test cover that and literally any scenario scenario. If the default ever changes or if this runs in parallel with any spot instance this test must always still remain green. This let us introducing any change while being confident we are not breaking the expectation.

> I think we need to simulate somewhere in the test suite that the daemonset is available and has more than 1 replica

Yes, I'll follow up with PRs to make sure the operator goes degraded if the pod crashloop and a test for it.

Reply:
Ack, SGTM